internal/status/registry.go

```diff
@@ -105,59 +106,65 @@ func (r Registration) Rolodex() string {
 func NewRegistry(_ context.Context, dir string) (*Registry, error) {
 	dir, err := filepath.Abs(dir)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("registry: filepath.Abs(%q): %w", dir, err)
+	}
+	if err = os.MkdirAll(dir, 0750); err != nil {
+		return nil, fmt.Errorf("registry: make dir %q: %w", dir, err)
 	}
-	err = os.MkdirAll(dir, 0750)
 	newClient := func(addr string) Server { return NewClient(addr) }
-	return &Registry{dir, newClient}, err
+	return &Registry{dir, newClient}, nil
 }
 
 // Register adds a registration to the registry.
 func (r *Registry) Register(ctx context.Context, reg Registration) error {
 	bytes, err := json.Marshal(reg)
 	if err != nil {
-		return err
+		return fmt.Errorf("registry: encode %v: %w", reg, err)
 	}
-	filename := fmt.Sprintf("%s.json", reg.DeploymentId)
-	w := files.NewWriter(filepath.Join(r.dir, filename))
+	filename := fmt.Sprintf("%s.registration.json", reg.DeploymentId)
```

Comment:
Does this change in filename require users to weaver purge the existing registrations?  Or will the existing registration files be simply ignored?

Reply:
I think the existing registrations will be ignored, but they could purge too. I'm still thinking about how to improve the whole purge-on-every-update situation we have right now. This change doesn't fix that.